test/utils/device.test.js

```diff
@@ -144,6 +144,25 @@ describe('device utility functions', () => {
     });
   });
 
+  describe('isTwiistLoop', () => {
+    it('should return `true` for a matching pattern within `client.name` and a major version above 2', () => {
+      const datum = { client: { name: 'com.sequelmedtech.tidepool-service', version: '2.0.0' } };
+      const datum2 = { client: { name: 'com.sequelmedtech.tidepool-service', version: '4.10.30' } };
+      expect(device.isTwiistLoop(datum)).to.be.true;
+      expect(device.isTwiistLoop(datum2)).to.be.true;
+    });
+
+    it('should return `false` for a non-matching pattern within `client.name`', () => {
+      const datum = { client: { name: 'com.tidepool.Loop', version: '2.0.0' } };
+      expect(device.isTwiistLoop(datum)).to.be.false;
+    });
+
+    it('should return `false` for a client major version below 2', () => {
+      const datum = { client: { name: 'com.tidepool.Loop', version: '1.9.9' } };
+      expect(device.isTwiistLoop(datum)).to.be.false;
+    });
```

Comment:
This test has the same `client.name` as the test for non-matching `client.name`, so may not actually be testing the major version restriction.

Reply:
Good catch :)  Fixing.